### DISCUSSION TITLE
RFC: Improve JS by converting to Typescript, commenting and minifying + sources RFC2021-04

### DISCUSSION THREAD

**Discussion by iJungleboy** [oqtane/oqtane.framework#1316] [Ideas]
ATM the JS in Oqtane is pure javascript and it has some bugs. 

If we would contribute changes chances are big that we need to put comments so future editors don't undo what we fix, but that would start to bloat the JS in unfriendly ways. 

The best practice as of now is IMHO to

1. Convert to TypeScript and split into multiple files, adding JSDocs etc.
2. Possibly even placing it in an own simple JS project in Visual Studio
3. Then building it with WebPack or TS and generating minified files + map files for debugging

We (2sic) have a lot of experience with this and would spearhead it if this is inline with the goals of Oqtane and would make it into the distribution. 

Your thoughts?

**Comment by sbwalker** [comment]
The general philosophy in Oqtane is to minimize the use of JavaScript as much as possible - as that is one of the primary benefits of Blazor. For those specific areas where JavaScript is required, Oqtane utilizes the standard JavaScript interop methodology recommended in Blazor. The goal for the JavaScript methods include in the framework was to avoid dependencies on jQuery or other frameworks and just rely on vanilla JavaScript. If there are bugs in the interop.js methods, please explain how to reproduce them and/or submit PRs to fix them.
